### PR TITLE
Avoid redownloading our own changes (fixes #144)

### LINKED ISSUE #144: Avoid redownloading changes on Sync
Refs: mozilla-services/readinglist#68

Basically, when doing a Batch before syncing, in the next update you will get all the records you've just changed that are already up-to-date with your local version. (When using ``If-Match`` headers to handle races)

We've just landed something, that let you exclude a list of values when syncing: https://github.com/mozilla-services/cliquet/pull/432

This is something we may want to use in kinto.js sync to optimize the second fetch.

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #144 

feedback? @Natim

### CODE REVIEW

File: src/collection.js [hunk 1]
```diff
@@ -727,10 +727,20 @@ export default class Collection {
       lastModified: this.lastModified,
       headers: {},
     }, options);
+
+    // Optionally ignore records when pulling for changes.
+    // (avoid redownloading our own changes on last step of #sync())
+    let filters;
+    if (options.exclude) {
+      // Take the first 50 records to limit size of querystring (~2000 chars).
```
Comment (n1k0) [inline src/collection.js]: Out of curiosity what is the actual length limitation of a query string?

Comment (leplatrem) [inline src/collection.js]: https://github.com/Kinto/kinto.js/issues/144#issuecomment-197787716

**Comment by Natim** [comment]
Seems good to me :)

**Comment by leplatrem** [comment]
Thanks! Merging!

**Pull request merged**